### PR TITLE
Implement SetSemanticFocus and Announce APIs 

### LINKED ISSUE #1711: [Spec] Add SetSemanticFocus and Announce methods for accessibility
### Description

Add SetSemanticFocus and Announce methods for accessibility. Both APIs have been introduce in .NET MAUI, and we want to make sure folks on Xamarin.Forms can leverage these new APIs as well.

`SetSemanticFocus` enables developers to indicate where they'd like the semantic focus (screen reader focus) to be sent.
`Announce` enables developers to indicate what they'd like the screen reader to announce, independently of controls.

SetSemanticFocus: https://github.com/dotnet/maui/pull/1829
Announce: https://github.com/dotnet/maui/pull/1760

### PR DESCRIPTION
**Pull request by rachelkang**
### Description of Bug ###
Implements SetSemanticFocus and Announce APIs for screen reader accessibility

Future TODO: Implement Support SetSemanticFocus on Windows

### Issues Fixed ###
- Fixes #1711

### Behavioral Changes ###

`SetSemanticFocus` can be leveraged to direct the screen reader focus to a specific control
`Announce` can be leveraged to instruct the screen reader to announce specific text

### PR Checklist ###

- [ ] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Semantic/SemanticExtensions.uwp.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+			var peer = FindAutomationPeer(Window.Current.Content);
+
+			// This GUID correlates to the internal messages used by UIA to perform an announce
+			// You can extract it  by using accessibility insights to monitor UIA events
+			// If you're curious how this works then do a google search for the GUID
+			peer?.RaiseNotificationEvent(
+				AutomationNotificationKind.ActionAborted,
+				AutomationNotificationProcessing.ImportantMostRecent,
+				text,
+				"270FA098-C644-40A2-A0BE-A9BEA1222A1E");
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Semantic/SemanticExtensions.uwp.cs:29]: could you move it to constants with an appropriate name?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Semantic/SemanticExtensions.uwp.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Media;
+
+namespace Xamarin.CommunityToolkit.Extensions
+{
+	public static partial class SemanticExtensions
+	{
+		static void PlatformSetSemanticFocus(this VisualElement element) =>
+			throw new NotSupportedException($"The current platform '{Device.RuntimePlatform}' does not support Xamarin.CommunityToolkit.SemanticExtensions");
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Semantic/SemanticExtensions.uwp.cs:12]: Note for the future me. Create an exception helper class like essentials

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Semantic/SemanticExtensions.uwp.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		}
+
+		// This isn't great but it's the only way I've found to announce with WinUI.
+		// You have to locate a control that has an automation peer and then use that 
+		// to perform the announce operation. This creates scenarios where the 
+		// screen might not have any automation peers on it to use but in those cases
+		// you really shouldn't be using the announce API
+		static AutomationPeer? FindAutomationPeer(DependencyObject depObj)
+		{
+			if (depObj != null)
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Semantic/SemanticExtensions.uwp.cs:39]: Can we invert this if?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Semantic/SemanticExtensions.uwp.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+			{
+				for (var i = 0; i < VisualTreeHelper.GetChildrenCount(depObj); i++)
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Semantic/SemanticExtensions.uwp.cs:41]: Can we cache the `VisualTreeHelper.GetChildrenCount()` value? (If it doesn't change)

**Review by pictos — COMMENTED** [review]
@rachelkang Thanks for this❣️ can you take a look on my suggestion on for loop? I'll keep it as a comment to not block the merge (:

**Review by pictos — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

**Pull request merged**